kroxylicious-schema-tools/kroxylicious-maven-plugin/src/main/java/io/kroxylicious/CompileSchemaMojo.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+import io.kroxylicious.tools.schema.compiler.FatalException;
+import io.kroxylicious.tools.schema.compiler.SchemaCompiler;
+
+@Mojo(name = "compile-schema", defaultPhase = LifecyclePhase.GENERATE_SOURCES)
+public class CompileSchemaMojo extends AbstractMojo {
+
+    @Parameter(defaultValue = "${project}")
+    private MavenProject project;
+
+    /**
+     * The input file or directory to be used for generating sources
+     *
+     */
+    @Parameter(property = "kroxylicious.schema.source", defaultValue = "${basedir}/src/main/schema")
+    File source;
+
+    @Parameter(property = "kroxylicious.schema.header")
+    File headerFile;
+
+    @Parameter(property = "kroxylicious.schema.dest", defaultValue = "${basedir}/target/generated-sources/java")
+    File target;
+
+    @Parameter(defaultValue = "false")
+    boolean failOnWarnings = false;
```

Comment:
```suggestion
    @Parameter(defaultValue = "true")
    boolean failOnWarnings = true;
```
I think we should default to getting warnings fixed.